include/nighthawk/common/termination_predicate.h

```diff
@@ -29,7 +29,7 @@ class TerminationPredicate {
    *
    * @param child the child predicate to link.
    */
-  virtual void link(TerminationPredicatePtr&& child) PURE;
+  virtual TerminationPredicate& link(TerminationPredicatePtr&& child) PURE;
```

Comment:
Please document `@return` in the comment above, not sure why this was needed, but the comment can explain :)

Reply:
Done